trillian/ctfe/storage/mysql/mysql.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package mysql defines the IssuanceChainStorage type, which implements IssuanceChainStorage interface with FindByKey and Add methods.
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"strings"
+
+	"k8s.io/klog/v2"
+
+	"github.com/go-sql-driver/mysql"
+)
+
+const (
+	selectIssuanceChainByKeySQL = "SELECT c.ChainValue FROM IssuanceChain AS c WHERE c.IdentityHash = ?"
+	insertIssuanceChainSQL      = "INSERT INTO IssuanceChain(IdentityHash, ChainValue) VALUES (?, ?)"
+)
+
+type IssuanceChainStorage struct {
+	db *sql.DB
+}
+
+// NewIssuanceChainStorage takes the database connection string as the input and return the IssuanceChainStorage.
+func NewIssuanceChainStorage(ctx context.Context, dbConn string) *IssuanceChainStorage {
+	conn := strings.Split(dbConn, "://")
+
+	db, err := open(ctx, conn[1])
+	if err != nil {
+		klog.Errorf("failed to open database: %v", err)
```

Comment:
Hmm, I think we should be consistent in behaviour in error cases here: if it's ok for `nil` to be returned then we should probably also return an error, or we can say that panicing here is preferable because if it's configured but impossible to create we can't start the service.

Reply:
Returning error in `open()` and panic in `NewIssuanceChainStorage()`.